### PR TITLE
Changed Object browser logic to work with websockets

### PR DESCRIPTION
**Pull request by bexsoft**
fixes https://github.com/minio/console/issues/943

## What does this do?

This allows us to start streaming results to the list instead of waiting to retrieve all the objects list before sending it to the client

Included a couple of fixes:

- Removed metadata for deleted items
- Fixed multiple metadata requests
- Fixed height grow for parent wrapper
- Fixed object reload after restore version

Signed-off-by: Benjamin Perez <benjamin@bexsoft.net>

### CODE REVIEW

File: restapi/ws_handle.go [hunk 1]
```diff
@@ -271,7 +285,48 @@ func serveWS(w http.ResponseWriter, req *http.Request) {
 			return
 		}
 		go wsAdminClient.profile(ctx, pOptions)
+	/*case strings.HasPrefix(wsPath, `/objects`):
```
Comment (jinapurapu) [inline restapi/ws_handle.go:288]: Should this be removed?

**Review by jinapurapu — CHANGES_REQUESTED** [review]

File: restapi/ws_handle.go (hunk 1 above)
Comment (bexsoft) [reply restapi/ws_handle.go]: Removed

**Comment by prakashsvmx** [comment]
@bexsoft  few observations while testing:
- Refresh button does not refresh the list ( mc cp an object to a path and click refresh- it is not reflected, hard reload reflects the object)
- Multiple metadata calls are made for a single object.
- meta data call for delete marker can be avoided?
<details>

https://user-images.githubusercontent.com/23444145/199458857-34dab3b9-f8e0-4737-8694-fe640af000f3.mp4

https://user-images.githubusercontent.com/23444145/199459610-adef957a-3cf5-4059-a948-0a0a6b672cc6.mp4



</details>

**Comment by kaankabalak** [comment]
@bexsoft When I go back from a path inside a bucket (`mybucket/mypath`) that has 20000 objects, to the bucket itself (`mybucket`), I still see the objects listed. Please see attached video:

https://user-images.githubusercontent.com/27314853/199496382-70b22351-13e5-49eb-a237-9a0035631cc9.mov

Steps to reproduce:

1- Start your MinIO Server. From this step onwards, the alias of the MinIO Server will be assumed to be named `myminio`
2- On your terminal instance, run the following commands to create a directory with 20000 files:
```
mkdir folder-test
cd folder-test
touch myfile{00001..20000}.txt 
```
3- Create a bucket on Console named `test`
4- Use the following command to upload the contents of the directory created in Step 2:
```
mc mirror folder-test myminio/test/folder
```
5- Login to the Console instance from your browser and navigate to the bucket named `test`
6- Navigate inside the path named `folder` and ensure that the objects are being listed.
7- Navigate back to the bucket named `test` using the breadcrumbs or the back button located next to it
8- Verify that the objects seen inside the path are also seen inside the bucket

**Comment by bexsoft** [comment]
> @bexsoft few observations while testing:
> 
> * Refresh button does not refresh the list ( mc cp an object to a path and click refresh- it is not reflected, hard reload reflects the object)
> * Multiple metadata calls are made for a single object.
> * meta data call for delete marker can be avoided?

Fixed

**Comment by bexsoft** [comment]
> @bexsoft When I go back from a path inside a bucket (`mybucket/mypath`) that has 20000 objects, to the bucket itself (`mybucket`), I still see the objects listed. Please see attached video:
> 
>  Screen.Recording.2022-11-02.at.4.01.09.PM.mov 
> Steps to reproduce:
> 
> 1- Start your MinIO Server. From this step onwards, the alias of the MinIO Server will be assumed to be named `myminio` 2- On your terminal instance, run the following commands to create a directory with 20000 files:
> 
> ```
> mkdir folder-test
> cd folder-test
> touch myfile{00001..20000}.txt 
> ```
> 
> 3- Create a bucket on Console named `test` 4- Use the following command to upload the contents of the directory created in Step 2:
> 
> ```
> mc mirror folder-test myminio/test/folder
> ```
> 
> 5- Login to the Console instance from your browser and navigate to the bucket named `test` 6- Navigate inside the path named `folder` and ensure that the objects are being listed. 7- Navigate back to the bucket named `test` using the breadcrumbs or the back button located next to it 8- Verify that the objects seen inside the path are also seen inside the bucket
> 
> Browser used: Google Chrome Version 106.0.5249.119

Improved request speed & added cancel on leave. Would you please help me to test again? Thank you in advance

**Review by prakashsvmx — APPROVED** [review]
Verified the fixes. Changes look good to me. :+1:

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**